Bug 2041940: Fix enablePortPoolsPrepopulation setting definition

MaysaMacedo opened the pull request:
As part of the work to reduce OpenStack resource usage[1]
the Namespace is only now handled when a Pod on Pods
Network is created in it. This new behavior also affects when
the ports pool prepopulation happens requiring update to
the docs.

[1] https://issues.redhat.com/browse/OSASINFRA-2590

MaysaMacedo commented:
/bugzilla refresh

dulek commented:
Looks good from Kuryr technical side.

gryf commented:
Yup, looks good.

MaysaMacedo commented:
/cc @itzikb-redhat 
as you were involved with the epic work.

MaysaMacedo commented:
/cc @maxwelldb

maxwelldb commented:
@MaysaMacedo Which OCP versions does this apply to?

MaysaMacedo commented:
> @MaysaMacedo Which OCP versions does this apply to?

@maxwelldb 4.10+

maxwelldb commented on modules/installation-osp-kuryr-port-pools.adoc:
```diff
@@ -17,7 +17,7 @@ Because Kuryr keeps each namespace in a separate subnet, a separate ports pool i
 Prior to installing a cluster, you can set the following parameters in the `cluster-network-03-config.yml` manifest file to configure ports pool behavior:
 
-* The `enablePortPoolsPrepopulation` parameter controls pool prepopulation, which forces Kuryr to add ports to the pool when it is created, such as when a new host is added, or a new namespace is created. The default value is `false`.
+* The `enablePortPoolsPrepopulation` parameter controls pool prepopulation, which forces Kuryr to add ports to the pools when the first pod on pods Network is created in a namespace. The default value is `false`.
```
@MaysaMacedo To be clear, which interpretation best matches this clause: "the first pod on pods Network"

1. The first pod on the network for pods
2. The first pod on the network resource for pods
3. The first pod on the network that a pod belongs to

maxwelldb replied:
```suggestion
* The `enablePortPoolsPrepopulation` parameter controls pool prepopulation, which forces Kuryr to add ports to pools when the first pod on the network for pods is created in a namespace. The default value is `false`.
```

maxwelldb replied:
The previous version has a grammar error, so I'm trying to figure out how to correct it by arriving at mutually agreeable phrasing. :+1: 

Beyond that, we use restrictive clauses when writing about cases like this in the docs, so something like "pod configured with..." becomes something like "pod that is configured with..."

"the first pod that is configured to use the [dedicated?] network for pods"?

MaysaMacedo replied:
ah ok, thanks for catching that. Yes, the latest version works for me:
`the first pod that is configured to use the dedicated network for pods`

maxwelldb replied:
```suggestion
* The `enablePortPoolsPrepopulation` parameter controls pool prepopulation, which forces Kuryr to add ports to the pools when the first pod that is configured to use the dedicated network for pods is created in a namespace. The default value is `false`.
```

maxwelldb commented on modules/installation-osp-kuryr-settings-active.adoc:
```diff
@@ -39,7 +39,7 @@ spec:
       poolMaxPorts: 5 <4>
 ----
-<1> Set `enablePortPoolsPrepopulation` to `true` to make Kuryr create new Neutron ports after a namespace is created or a new node is added to the cluster. This setting raises the Neutron ports quota but can reduce the time that is required to spawn pods. The default value is `false`.
+<1> Set `enablePortPoolsPrepopulation` to `true` to make Kuryr create new Neutron ports when the first pod on pods network in a namespace is created. This setting raises the Neutron ports quota but can reduce the time that is required to spawn pods. The default value is `false`.
```
```suggestion
<1> Set `enablePortPoolsPrepopulation` to `true` to make Kuryr create new Neutron ports when the first pod on the network for pods is created in a namespace. This setting raises the Neutron ports quota but can reduce the time that is required to spawn pods. The default value is `false`.
```

maxwelldb commented on modules/installation-osp-kuryr-settings-installing.adoc:
```diff
@@ -80,7 +80,7 @@ spec:
       openstackServiceNetwork: 172.30.0.0/15 <5>
 ----
-<1> Set the value of `enablePortPoolsPrepopulation` to `true` to make Kuryr create new Neutron ports after a namespace is created or a new node is added to the cluster. This setting raises the Neutron ports quota but can reduce the time that is required to spawn pods. The default value is `false`.
+<1> Set the value of `enablePortPoolsPrepopulation` to `true` to make Kuryr create new Neutron ports when the first pod on pods network in a namespace is created. This setting raises the Neutron ports quota but can reduce the time that is required to spawn pods. The default value is `false`.
```
```suggestion
<1> Set `enablePortPoolsPrepopulation` to `true` to make Kuryr create new Neutron ports when the first pod on the network for pods is created in a namespace. This setting raises the Neutron ports quota but can reduce the time that is required to spawn pods. The default value is `false`.
```

maxwelldb commented on rest_api/operator_apis/network-operator-openshift-io-v1.adoc:
```diff
@@ -494,7 +494,7 @@ Type::
 | `enablePortPoolsPrepopulation`
 | `boolean`
-| enablePortPoolsPrepopulation when true will make Kuryr prepopulate each newly created port pool with a minimum number of ports. Kuryr uses Neutron port pooling to fight the fact that it takes a significant amount of time to create one. Instead of creating it when pod is being deployed, Kuryr keeps a number of ports ready to be attached to pods. By default port prepopulation is disabled.
+| enablePortPoolsPrepopulation when true will make Kuryr prepopulate each newly created port pool with a minimum number of ports. Kuryr uses Neutron port pooling to fight the fact that it takes a significant amount of time to create one. It creates a number of ports when the first pod on pods network is being deployed in a namespace and keeps them ready to be attached to pods. By default port pre ...
```
```suggestion
| enablePortPoolsPrepopulation when true will make Kuryr prepopulate each newly created port pool with a minimum number of ports. Kuryr uses Neutron port pooling to fight the fact that it takes a significant amount of time to create one. It creates a number of ports when the first pod on the network for pods is being deployed in a namespace, and keeps them ready to be attached to pods. Port prepopulation is disabled by default.
```

maxwelldb reviewed (commented):
Made some suggestions. @MaysaMacedo

maxwelldb commented:
@MaysaMacedo LMK if you're cool with the suggestions in the comments. If so, we can commit them and then this can go to QE.

MaysaMacedo commented:
@maxwelldb looks good to me. Thanks!

maxwelldb commented:
@jboxman Maysa's looking to update content that is in the API docs. I haven't had to do that before. What's the best way to make sure that her changes stick?

maxwelldb commented:
@MaysaMacedo It may be that you'll need to make the API docs change outside of the docs repo--probably to: https://github.com/openshift/api/

What looks like API docs on our side are actually generated from there.

MaysaMacedo commented:
@maxwelldb okay, I removed the API docs and proposed the changes against [API repo](https://github.com/openshift/api/pull/1132). I have also squashed the commits.

maxwelldb commented:
@MaysaMacedo Thanks!

maxwelldb commented:
@eurijon Can you +1 this when you have a moment?

kalexand-rh commented on modules/installation-osp-kuryr-settings-active.adoc:
```diff
@@ -39,7 +39,7 @@ spec:
       poolMaxPorts: 5 <4>
 ----
-<1> Set `enablePortPoolsPrepopulation` to `true` to make Kuryr create new Neutron ports after a namespace is created or a new node is added to the cluster. This setting raises the Neutron ports quota but can reduce the time that is required to spawn pods. The default value is `false`.
+<1> Set `enablePortPoolsPrepopulation` to `true` to make Kuryr create new Neutron ports when the first pod that is configured to use the dedicated network for pods is created in a namespace. This setting raises the Neutron ports quota but can reduce the time that is required to spawn pods. The default value is `false`.
```
~new~

Also, sometimes you call them both "ports" and "Neutron ports." Does the extra info make sense in only some contexts?

maxwelldb replied:
^^^ Yes. Not all ports in ShiftStack docs are Neutron-related.

maxwelldb replied:
```suggestion
<1> Set `enablePortPoolsPrepopulation` to `true` to make Kuryr create Neutron ports when the first pod that is configured to use the dedicated network for pods is created in a namespace. This setting raises the Neutron ports quota but can reduce the time that is required to spawn pods. The default value is `false`.
```

maxwelldb commented:
@MaysaMacedo Looks good. If you can just squash this down to one commit, we can merge this after @eurijon +1s it.

MaysaMacedo commented:
@maxwelldb done

itzikb-redhat commented:
LGTM

MaysaMacedo commented:
@maxwelldb do you think this one is ready?

eurijon commented:
@maxwelldb it lgtm too

maxwelldb commented:
@MaysaMacedo The main and enterprise-4.10 branches are frozen until GA. I can merge this after they're unfrozen, and the change will be reflected in the 4.10 docs that same day.

bobfuru commented:
In preparation for OCP 4.10 GA, I'm moving the Milestone to "Next Release" as this PR did not make the merge by the 4.10 GA date. Post-GA, any open 4.10 PRs fall under the "Next Release" bucket in the same way that 4.6-4.9 PRs are already there. This change does not have any impact on the work in this PR; it's a housekeeping task to keep account of all PRs that had already merged by the 4.10 GA date. 😁

maxwelldb commented:
@MaysaMacedo Any objections to my merging this?

MaysaMacedo commented:
@maxwelldb No

maxwelldb commented:
/cherry-pick enterprise-4.10

maxwelldb commented:
/cherry-pick enterprise-4.11

maxwelldb commented:
@MaysaMacedo Done. Feel free to adjust your related BZ to reflect this. You should see your changes in prod today.
